packages/web3-eth-contract/src/contract.ts

```diff
@@ -114,7 +114,7 @@ type ContractBoundMethod<
 	Abi extends AbiFunctionFragment,
 	Method extends ContractMethod<Abi> = ContractMethod<Abi>,
 > = (
-	...args: Method['Inputs']
+	...args: Abi extends AbiFunctionFragment ? Method['Inputs'] : any
```

Comment:
we are sure about correct types and encoding based on provided ABI instead of using only param types in ts directly. so i'll suggest to use ABI. and we dnt add feature for "accept parameters even when no ABI was provided"

Reply:
Sorry, for the confusion. I now edited the title of the PR to describe the most actual practical use case. And also added elaboration in the PR description.

And many thanks for reviewing this PR even it is still in draft.